plugins/com.google.cloud.tools.eclipse.appengine.libraries.ui/src/com/google/cloud/tools/eclipse/appengine/libraries/ui/CloudLibrariesSelectionPage.java

```diff
@@ -0,0 +1,120 @@
+import org.eclipse.jface.viewers.StructuredSelection;
+import org.eclipse.jface.wizard.WizardPage;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Composite;
+
+/**
+ * UI to select useful libraries for GCP projects
+ */
+public class CloudLibrariesSelectionPage extends WizardPage {
+
+  /**
+   * The library groups to be displayed; pairs of (id, title). For example, <em>"clientapis" &rarr;
+   * "Google Cloud APIs for Java"</em>.
+   */
+  @VisibleForTesting
+  Map<String, String> libraryGroups;
+  protected final List<LibrarySelectorGroup> librariesSelectors = new ArrayList<>();
+  
+  /** The initially selected libraries. */
+  protected List<Library> initialSelection = Collections.emptyList();
+
+
+  public CloudLibrariesSelectionPage() {
+    super("cloudPlatformLibrariesPage"); //$NON-NLS-1$
+    setTitle(Messages.getString("cloud-platform-libraries-title")); //$NON-NLS-1$
+    setDescription(Messages.getString("apiclientlibrariespage-description")); //$NON-NLS-1$
+    setImageDescriptor(SharedImages.GCP_WIZARD_IMAGE_DESCRIPTOR);
+  }
+
+  /**
+   * Return the list of selected libraries.
+   */
+  protected final List<Library> getSelectedLibraries() {
+    List<Library> selectedLibraries = new ArrayList<>();
+    for (LibrarySelectorGroup librariesSelector : librariesSelectors) {
+      selectedLibraries.addAll(librariesSelector.getSelectedLibraries());
+    }
+    return selectedLibraries;
+  }
+  
+  protected final void setSelectedLibraries(Collection<Library> selectedLibraries) {
+    initialSelection = new ArrayList<>(selectedLibraries);
+    if (!librariesSelectors.isEmpty()) {
+      for (LibrarySelectorGroup librarySelector : librariesSelectors) {
+        librarySelector.setSelection(new StructuredSelection(initialSelection));
+      }
+    }
+  }
+
+  /**
+   * Set the different library groups to be shown; must be called before controls are created.
+   */
+  protected final void setLibraryGroups(Map<String, String> groups) {
+    libraryGroups = groups;
+  }
+  
+  /**
+   * The libraries that are available for selection.
+   */
+  protected final List<Library> getAvailableLibraries() {
+    List<Library> available = new ArrayList<>();
+    for (String libraryGroupId : libraryGroups.keySet()) {
+      available.addAll(CloudLibraries.getLibraries(libraryGroupId));
+    }
+    return available;
+  }
+  
+  @Override
+  public void createControl(Composite parent) {
+    createWidget(parent, false);
+  }
+
+  protected final void createWidget(Composite parent, boolean java7) {
+    Preconditions.checkNotNull(libraryGroups, "Library groups must be set"); //$NON-NLS-1$
+    Composite composite = new Composite(parent, SWT.NONE);
+
+    // create the library selector libraryGroups
+    for (Entry<String, String> group : libraryGroups.entrySet()) {
+      LibrarySelectorGroup librariesSelector =
+          new LibrarySelectorGroup(composite, group.getKey(), group.getValue(), java7);
```

Comment:
The `LibrarySelectorGroup` constructor uses the name `restrictedEnvironment` for the parameter `java7` that sounds more pleasant. `java7` which is too explicit and narrow.

Reply:
done